src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1106,15 +1106,21 @@ bool PolicyHandler::ReceiveMessageFromSDK(const std::string& file,
                                           const BinaryMessage& pt_string) {
   POLICY_LIB_CHECK(false);
 
-  bool ret = policy_manager_->LoadPT(file, pt_string);
-  LOG4CXX_INFO(logger_, "Policy table is saved: " << std::boolalpha << ret);
-  if (ret) {
+  auto load_pt_result = policy_manager_->LoadPT(file, pt_string);
+
+  LOG4CXX_INFO(logger_, "Load policy table result code: " << load_pt_result);
+
+  const bool is_ptu_successful =
+      load_pt_result == PolicyManager::LoadPtResult::kSuccess;
+  OnPTUFinished(is_ptu_successful);
+
+  if (is_ptu_successful) {
     LOG4CXX_INFO(logger_, "PTU was successful.");
     policy_manager_->CleanupUnpairedDevices();
-    int32_t correlation_id = application_manager_.GetNextHMICorrelationID();
-
     SetDaysAfterEpoch();
+    policy_manager_->OnPTUFinished(load_pt_result);
```

Comment:
@AKalinich-Luxoft is there any reason, for not doing this once before or after `if (is_ptu_successful)`?

Reply:
@mkorniichuk yes. The main idea of this PR was to provide a strict order of policy actions and send UP_TO_DATE only when all policy related actions were done.
This is the best place for `policy_manager_->OnPTUFinished` as such functions as `CleanupUnpairedDevices` and `SetDaysAfterEpoch` are still affecting policy database.